pkg/controller/metrics/metrics_test.go

```diff
@@ -46,47 +46,87 @@ func TestCredentialsRequests(t *testing.T) {
+				// provision failed condition
+				testCredReqWithConditions(testAWSCredRequest("aprovisionfailed"), []credreqv1.CredentialsRequestCondition{provisionFailedCond}),
+				// provision failed false condition
+				func() credreqv1.CredentialsRequest {
+					cr := testCredReqWithConditions(testAWSCredRequest("aprovisionnotfailed"), []credreqv1.CredentialsRequestCondition{provisionFailedCond})
+					cr.Status.Conditions[0].Status = corev1.ConditionFalse
+					return cr
+				}(),
+				// insufficient cloud creds condition
+				testCredReqWithConditions(testAWSCredRequest("ainsufficientcreds"), []credreqv1.CredentialsRequestCondition{insufficientCredsCond}),
+
+				// regular GCP credreq
+				testGCPCredRequest("gregular"),
+				// GCP credreq with condition set
+				testCredReqWithConditions(testGCPCredRequest("gignored"), []credreqv1.CredentialsRequestCondition{ignoredCond}),
+			},
+			validate: func(accumulator *credRequestAccumulator, t *testing.T) {
+				// total cred requests
+				assert.Equal(t, 5, accumulator.crTotals["aws"])
+				assert.Equal(t, 2, accumulator.crTotals["gcp"])
+
+				// conditions
+				assert.Equal(t, 1, accumulator.crConditions[credreqv1.MissingTargetNamespace])
+				assert.Equal(t, 1, accumulator.crConditions[credreqv1.CredentialsProvisionFailure])
+				assert.Equal(t, 1, accumulator.crConditions[credreqv1.Ignored])
+				assert.Equal(t, 1, accumulator.crConditions[credreqv1.InsufficientCloudCredentials])
+			},
+		},
+		{
+			name:        "cco disabled report no conditions",
+			ccoDisabled: true,
+			credReqs: []credreqv1.CredentialsRequest{
+				// missing namespace condition
+				testCredReqWithConditions(testAWSCredRequest("amissingnamespace"), []credreqv1.CredentialsRequestCondition{missingTargetNamespaceCond}),
+				// provision failed condition
+				testCredReqWithConditions(testAWSCredRequest("aprovisionfailed"), []credreqv1.CredentialsRequestCondition{provisionFailedCond}),
+				// insufficient cloud creds condition
+				testCredReqWithConditions(testAWSCredRequest("ainsufficientcreds"), []credreqv1.CredentialsRequestCondition{insufficientCredsCond}),
+
+				// GCP credreq with condition set
+				testCredReqWithConditions(testGCPCredRequest("gignored"), []credreqv1.CredentialsRequestCondition{ignoredCond}),
+			},
+			validate: func(accumulator *credRequestAccumulator, t *testing.T) {
+				// total cred requests
+				assert.Equal(t, 3, accumulator.crTotals["aws"])
+				assert.Equal(t, 1, accumulator.crTotals["gcp"])
+
+				// conditions should all be zero as CCO is disabled
+				assert.Equal(t, 0, accumulator.crConditions[credreqv1.MissingTargetNamespace])
+				assert.Equal(t, 0, accumulator.crConditions[credreqv1.CredentialsProvisionFailure])
+				assert.Equal(t, 0, accumulator.crConditions[credreqv1.Ignored])
+				assert.Equal(t, 0, accumulator.crConditions[credreqv1.InsufficientCloudCredentials])
+			},
+		},
```

Comment:
Just a minor style comment, sub-test here probably isn't buying you much, seems there's almost no shared code and the validation functions are separate. Fine as it is though!